src/main/java/core/basesyntax/service/CarServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.CarDao;
+import core.basesyntax.lib.Inject;
+import core.basesyntax.lib.Service;
+import core.basesyntax.model.Car;
+import core.basesyntax.model.Driver;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CarServiceImpl implements CarService {
+    @Inject
+    private CarDao carDao;
+
+    @Override
+    public Car create(Car car) {
+        return carDao.create(car);
+    }
+
+    @Override
+    public Car get(Long id) {
+        return carDao.get(id).get();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return carDao.getAll();
+    }
+
+    @Override
+    public Car update(Car car) {
+        return carDao.update(car);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        carDao.update(car).getDrivers().add(driver);
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        carDao.update(car).getDrivers().remove(driver);
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> driverCars = new ArrayList<>();
+        for (Car car : carDao.getAll()) {
+            for (Driver driver : car.getDrivers()) {
+                if (driver.getId().equals(driverId)) {
+                    driverCars.add(car);
+                }
```

Comment:
move this logic to dao layer, we should receive filtered data from dao

Reply:
done